### PR TITLE
Simplify obtention of forward and backward masks in trajectory balance loss

### PR DESCRIPTION
**Pull request by alexhernandezgarcia**
The main goal of this PR is to avoid the re-computation of invalid masks for the forward and backward logits:
* The masks of the forward logits (for the parents) are obtained from the (states) masks already included in the batch.
* The masks of the backward logits (fo the states) are obtained from the parents actions already included in the batch.

I have removed the legacy code from the previous PR.

### CODE REVIEW

**Comment by nikita-0209** [comment]
makes sense to me
